### PR TITLE
Added redux-recompose and refactored accordingly

### PR DESCRIPTION
**Pull request by hernan-clich**
## Summary

Added redux-recompose and refactored accordingly

## Screenshots

There were no noticeable changes in the UI.

### Reactotron successfully logs dispatched actions after the setup and data is fetched and properly displayed on screen:

rn_redux_recompose

## Trello Card

https://trello.com/c/CLv1yReY/14-n-96-redux-recompose-i-teor%C3%ADa
https://trello.com/c/iGnN7mWS/23-n-97-redux-recompose-ii-refactor-book-reducer

### CODE REVIEW

File: wbooks/src/interfaces/bookState.ts [hunk 1]
```diff
@@ -2,6 +2,4 @@ import { Book } from '@interfaces/books';
 
 export interface BookState {
   books: Book[];
-  error: string;
-  loading: boolean;
```
Comment (sfernandez11) [inline wbooks/src/interfaces/bookState.ts:6]: Are you sure? What happens with the `completeState` in the reducer?

File: wbooks/src/interfaces/rootState.ts [hunk 2]
```diff
@@ -1,3 +1,8 @@
-import { rootReducer } from '@redux/store';
+import { $CombinedState } from 'redux';
+import { BookState } from '@interfaces/bookState';
 
-export type RootState = ReturnType<typeof rootReducer>;
+export type RootState = {
+  readonly [$CombinedState]?: undefined;
+} & {
+  books: BookState;
+};
```
Comment (sfernandez11) [inline wbooks/src/interfaces/rootState.ts]: Why all of this complex logic for an interface?
```
export interface RootState = {
  books: BookState;
}
```

File: wbooks/src/redux/book/reducer.ts [hunk 3]
```diff
@@ -1,25 +1,16 @@
...
 
 import { actions } from './actions';
 
-const initialState: BookState = {
-  books: [],
-  error: '',
-  loading: false
+const initialState: BookState = completeState({
+  books: []
+});
```
Comment (sfernandez11) [inline wbooks/src/redux/book/reducer.ts]: Wrong, check again the [docs](https://github.com/Wolox/redux-recompose/tree/master/src/completers/completeState)

**Review by sfernandez11 — CHANGES_REQUESTED** [review]

File: wbooks/src/interfaces/bookState.ts (hunk 1 above)
Comment (hernan-clich) [reply wbooks/src/interfaces/bookState.ts:6]: You're right, I'm now considering `booksLoading` and `booksError`

File: wbooks/src/interfaces/rootState.ts (hunk 2 above)
Comment (hernan-clich) [reply wbooks/src/interfaces/rootState.ts]: You're right, I thought we had to consider that all reducers are wrapped with redux's `combineReducers` function, apparently not the case, good to know.

File: wbooks/src/redux/book/reducer.ts (hunk 3 above)
Comment (hernan-clich) [reply wbooks/src/redux/book/reducer.ts]: I'm now wrapping it with `description: { ... }`

**Review by sfernandez11 — APPROVED** [review]
🥇

**Pull request merged**